.github/workflows/precommit.yaml

```diff
@@ -0,0 +1,47 @@
+name: "Precommit tests"
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/precommit.yaml'
+      - 'llvm/**'
+
+jobs:
+  build-llvm:
+    name: "Build and test LLVM"
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/llvm/ci-ubuntu-22.04:latest
+    steps:
+      - name: Fetch LLVM sources
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+      - name: Setup ccache
+        uses: hendrikmuhs/ccache-action@v1
+        with:
+          max-size: 500M
+          variant: sccache
+          key: precommit-linux
```

Comment:
How well do you think the caching will work if we aren't running this job post-commit.

Reply:
I think it will work reasonably well. The job should be run pretty often to keep a warm cache. Things might be a little bit less optimal if there are PRs that touch common headers that trigger a lot of rebuilt files or something and things go back and forth, but I think with a reasonable cache size, we should be fine. The max size is 2-3x the size of the cache generated from a cold build from my experiments, so we should be able to hold a reasonable amount of objects.

I think this is something that theoretically should work, but we'll just have to evaluate in practice however.